File: Comunicación entre formularios Windows Forms/[VB]-Comunicación entre formularios Windows Forms/VB/MainForm.vb
Public Class MainForm

    Private Sub MainForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        dataGridView1.Rows.Add("Darth", "Vader", True)
        dataGridView1.Rows.Add("Luke", "Skywalker", False)
        dataGridView1.Rows.Add("Leia", "Organa", True)
        dataGridView1.Rows.Add("Han", "Solo", False)
        dataGridView1.Rows.Add("Obi-Wan", "Kenobi", True)
        dataGridView1.Rows.Add("Wilhuff", "Tarkin", False)
    End Sub

    Private Sub dataGridView1_CellContentClick(sender As Object, e As DataGridViewCellEventArgs) Handles dataGridView1.CellContentClick
        Dim grid = CType(sender, DataGridView)
        If grid.Columns(e.ColumnIndex).Name <> "EditColumn" Then
            Return
        End If

        Dim rowToEdit = grid.Rows(e.RowIndex)
        Dim dialog = New DialogForm() With
        {
            .CurrentPerson = New Person() With
            {
                .Name = rowToEdit.Cells("NameColumn").Value,
                .Surname = rowToEdit.Cells("SurnameColumn").Value,
                .Active = rowToEdit.Cells("ActiveColumn").Value
            } 
        }

        If dialog.ShowDialog() = DialogResult.OK Then
            rowToEdit.Cells("NameColumn").Value = dialog.CurrentPerson.Name
            rowToEdit.Cells("SurnameColumn").Value = dialog.CurrentPerson.Surname
            rowToEdit.Cells("ActiveColumn").Value = dialog.CurrentPerson.Active
        End If
    End Sub

    Private Sub btnAdd_Click(sender As Object, e As EventArgs) Handles btnAdd.Click
        Dim dialog = New DialogForm()
        If dialog.ShowDialog() = DialogResult.OK Then
            dataGridView1.Rows.Add(
                dialog.CurrentPerson.Name,
                dialog.CurrentPerson.Surname,
                dialog.CurrentPerson.Active
            )
        End If
    End Sub

End Class

File: Comunicación entre formularios Windows Forms/[C#]-Comunicación entre formularios Windows Forms/C#/MainForm.cs
using System;
using System.Windows.Forms;

namespace DialogForm
{
    public partial class MainForm : Form
    {
        public MainForm()
        {
            InitializeComponent();
        }


        private void MainForm_Load(object sender, EventArgs e)
        {
            dataGridView1.Rows.Add("Darth", "Vader", true);
            dataGridView1.Rows.Add("Luke", "Skywalker", false);
            dataGridView1.Rows.Add("Leia", "Organa", true);
            dataGridView1.Rows.Add("Han", "Solo", false);
            dataGridView1.Rows.Add("Obi-Wan", "Kenobi", true);
            dataGridView1.Rows.Add("Wilhuff", "Tarkin", false);
        }

        private void dataGridView1_CellContentClick(object sender, DataGridViewCellEventArgs e)
        {
            var grid = (DataGridView) sender;
            if (grid.Columns[e.ColumnIndex].Name != "EditColumn") return;

            var rowToEdit = grid.Rows[e.RowIndex];
            var dialog = new DialogForm
            {
                CurentPerson = new Person()
                {
                    Name = (string) rowToEdit.Cells["NameColumn"].Value,
                    Surname = (string) rowToEdit.Cells["SurnameColumn"].Value,
                    Active = (bool) rowToEdit.Cells["ActiveColumn"].Value
                }
            };

            if (dialog.ShowDialog() == DialogResult.OK)
            {
                rowToEdit.Cells["NameColumn"].Value = dialog.CurentPerson.Name;
                rowToEdit.Cells["SurnameColumn"].Value = dialog.CurentPerson.Surname;
                rowToEdit.Cells["ActiveColumn"].Value = dialog.CurentPerson.Active;
            }
        }

        private void btnAdd_Click(object sender, EventArgs e)
        {
            var dialog = new DialogForm();
            if (dialog.ShowDialog() == DialogResult.OK)
            {
                dataGridView1.Rows.Add(
                    dialog.CurentPerson.Name,
                    dialog.CurentPerson.Surname,
                    dialog.CurentPerson.Active
                    );
            }
        }
    }
}
